components/src/core/Drawer/DrawerRailItem.tsx

```diff
@@ -0,0 +1,259 @@
+        onItemSelect,
+        ripple = true,
+        title,
+        /* eslint-disable @typescript-eslint/no-unused-vars */
+        activeItemBackgroundColor,
+        activeItemFontColor,
+        activeItemIconColor,
+        divider,
+        itemFontColor,
+        itemIconColor,
+        statusColor,
+        /* eslint-disable @typescript-eslint/no-unused-vars */
+
+        ...buttonBaseProps
+    } = props;
+
+    const defaultClasses = useStyles(props);
+
+    const active = activeItem === itemID;
+    const hasAction = Boolean(onItemSelect || onClick);
+
+    // Customize the color of the Touch Ripple
+    const RippleProps =
+        ripple && hasAction
+            ? {
+                  TouchRippleProps: {
+                      classes: {
+                          child: defaultClasses.ripple,
+                      },
+                  },
+              }
+            : {};
+
+    const combine = useCallback(
+        (className: keyof DrawerRailItemClasses): string => clsx(defaultClasses[className], classes[className]),
+        [defaultClasses, classes]
+    );
+
+    const getIcon = useCallback((): JSX.Element | undefined => {
+        if (icon) {
+            return <Avatar className={combine('icon')}>{icon}</Avatar>;
+        }
+    }, [icon, combine]);
+
+    const onClickAction = useCallback(
+        (e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+            if (onItemSelect) {
+                onItemSelect();
+            }
+            if (onClick) {
+                onClick(e);
+            }
+        },
+        [onItemSelect, onClick]
+    );
+
+    const innerContent = (
+        <ButtonBase
+            ref={ref}
+            {...buttonBaseProps}
+            className={clsx(defaultClasses.root, classes.root, {
+                [defaultClasses.condensed]: condensed,
+                [classes.condensed]: condensed,
+            })}
+            disableRipple={!ripple || !hasAction}
+            onClick={onClickAction}
+            {...RippleProps}
+        >
+            {/* Active Item Highlight */}
+            {active && <div className={combine('active')} />}
+            {/* Status Color Stripe */}
+            <div className={combine('statusStripe')} data-test={'status-stripe'} />
+            {/* Icon */}
+            {getIcon()}
+            {/* Title */}
+            {!condensed && (
+                <Typography
+                    variant={'caption'}
+                    className={clsx(defaultClasses.title, classes.title, {
+                        [defaultClasses.titleActive]: active,
+                        [classes.titleActive]: active,
+                    })}
+                >
+                    {title}
+                </Typography>
+            )}
+            {/* Divider */}
+            {props.divider && <Divider className={combine('divider')} />}
+        </ButtonBase>
+    );
+
+    return hidden ? null : condensed ? (
+        <Tooltip title="Add" placement="right">
```

Comment:
I agree it would be nice to be able to turn it off...trying to think of the best way to do that. I'm hesitant to add a prop that is new (i.e. not supported by the NavItem), because I have to find some way to pass it in that's not through the `items` prop, or do the battle with types we're talking about below.

We could add a `showTooltip` prop to the NavItem type and implement (optional) tooltips for regular drawer items and rail items. Thoughts?

Reply:
I'm inclined to leave the prop out for now (since we currently match Angular) and I can make an enhancement to discuss if we want to give users the ability to enable/disable tooltips and implement it in both frameworks. Is that cool?

Reply:
https://github.com/pxblue/react-component-library/issues/223